toponetx/classes/cell_complex.py

```diff
@@ -1979,14 +2106,50 @@ def clone(self) -> "CellComplex":
             CX.add_cell(cell.clone())
         return CX
 
+    def euler_characterisitics(self) -> int:
+        """Euler characterisitics of the cell complex."""
+        return len(self.nodes) - len(self.edges) + len(self.cells)
+
     def remove_singletons(self) -> "CellComplex":
         """Remove singleton nodes (see `CellComplex.singletons()`)."""
         for node in self.singletons():
             self._G.remove_node(node)
 
-    def s_connected_components(
-        self, s: int = 1, cells: bool = True, return_singletons: bool = False
-    ):
+    def get_linegraph(self, s=1, cells=True):
+        """Create line graph of self.
+
+        If cells=True (default), the cells will be the vertices of the line graph.
+        Two vertices are connected by an s-line-graph edge if the corresponding cell
+        complex edges intersect in at least s cell complex nodes.
+
+        If cells=False, the cell complex nodes will be the vertices of the line graph.
+        Two vertices are connected if the nodes they correspond to share at least s
+        incident cell complex edges.
+
+        Parameters
+        ----------
+        s : int
+            The width of the connections.
+        cells : bool, optional, default=True
+            Determines if cells or nodes will be the vertices in the line graph.
+
+        Returns
+        -------
+        nx.Graph
+            A NetworkX graph representing the s-linegraph of the Cell Complex.
+        """
+        if isinstance(s, None):
+            ValueError(
+                "s must be a positive integer larger than 1, got type of s None."
+            )
+        if cells:
+            M = self.all_cell_to_node_codjacnecy_matrix(s=s)
+        else:
+            M = self.node_to_all_cell_adjacnecy_matrix(s=s)
+
+        return nx.from_scipy_sparse_array(M)
+
+    def s_connected_components(self, s=1, cells=True, return_singletons=False):
```

Comment:
Please add the type hints back (for some reason GitHub shows the old line somewhat further above)